### PR TITLE
implement #15 

### PR DESCRIPTION
**Pull request by junnmm**
This pull request implements #15, which is a simulator for the issue mentioned in https://github.com/kcc-community/kcc/pull/82.  

This pull request comes with a file named `chain.tar.gz`, which is an exported chain of more than 110000 blocks.  After importing the blocks from this file, the chain's ancient db will not be empty. However, there are not any transactions in these blocks, which means the ancient db will not have any receipts in it.

### CODE REVIEW

File: simulators/kcc/ancient-db/geth.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+# Don't immediately abort, some imports are meant to fail
+set +e
+
+
+# uncompress the exported chain 
+if [ -f /chain.tar.gz ]; then
+    echo "Uncompressing chain.tar.gz..."
+    tar -xvzf /chain.tar.gz -C /
+fi
```
Comment (junnmm) [inline simulators/kcc/ancient-db/geth.sh:106]: The `chain.tar.gz` file contains a chain of more than 110000 blocks.

File: simulators/kcc/ancient-db/geth.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+
+# Load the test chain if present
+echo "Loading initial blockchain..."
+if [ -f /chain.rlp ]; then
+    $geth $FLAGS --gcmode=archive import /chain.rlp
+else
+    echo "Warning: chain.rlp not found."
+fi
```
Comment (junnmm) [inline simulators/kcc/ancient-db/geth.sh:115]: Import the blocks from `chain.rlp` into our local chain.

File: simulators/kcc/ancient-db/main.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+func runChain(t *hivesim.T, c *hivesim.Client) {
+
+	// We need to import exported blocks into the chain before starting it.
+	// However, to import more than 110000 blocks into the chain and compact them would cost more than 3 minutes.
+	// Remember to start `hive` with an extra option of `--client.checktimelimit 10m` to avoid timeout.
+
+	// All the 110000 blocks are empty. So there are no receipts in the ancient db.
+
+	// Why 110000? Because we will only have ancient items if the block number is greater than 90000.
```
Comment (junnmm) [inline simulators/kcc/ancient-db/main.go:70]: Importing 110000 blocks would cost a few minutes, hive should allow such a long time to start.

File: simulators/kcc/ancient-db/main.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		Description: "Testcase for ancient db",
+	}
+	suite.Add(hivesim.ClientTestSpec{
+		Role:        "eth1",
+		Name:        "Run the chain",
+		Description: "Run a chain to generate ancient db",
+		Files: map[string]string{
+			"/genesis.json": "genesis.json",
+			"/chain.tar.gz": "chain.tar.gz", // Add our pre exported chain
+			"/geth.sh":      "geth.sh",      // override the default geth.sh
```
Comment (junnmm) [inline simulators/kcc/ancient-db/main.go:30]: Mount several files into the container.

**Review by newgoo — COMMENTED** [review]
looks good to me.

**Pull request merged**